File: curp_ws/clientesws/WsCurpConsultaClient/src/mx/gob/renapo/ecurp/ws/gestion/cliente/DatosConsultaDetalles.java
package mx.gob.renapo.ecurp.ws.gestion.cliente;
/**
 *
 * Clase que actúa como Bean y encapsula los datos que se envían como parámetros para
 * realizar la consulta por detalles.
 *
 */

//package mx.gob.segob.dgti.ecurp.wserv.services;

/**
 * @author Dalma Casas Vargas
 * 
 * TODO To change the template for this generated type comment go to Window -
 * Preferences - Java - Code Style - Code Templates
 */
public class DatosConsultaDetalles {
    private int tipoTransaccion;

    /**
     * <code>cveUsuario</code> Usuario
     */
    private String cveUsuario;

    /**
     * <code>password</code> Password
     */
    private String password;

    /**
     * <code>direccionIp</code> Dirección IP
     */
    private String direccionIp;

    /**
     * <code>nombre</code> Nombre
     */
    private String nombre;

    /**
     * <code>primerApellido</code> Primer apellido
     */
    private String primerApellido;

    /**
     * <code>fechaNacimiento</code> Fecha de nacimiento
     */
    private String fechaNacimiento;

    /**
     * <code>segundoApellido</code> Segundo apellido
     */
    private String segundoApellido;

    /**
     * <code>sexo</code> Sexo
     */
    private String sexo;

    /**
     * <code>cveAlfaEntFedNac</code> Entidad federativa de nacimiento
     */
    private String cveAlfaEntFedNac;
    
    /**
     * <code>cveEntidadEmisora</code> Clave de la entidad emisora
     */
    private String cveEntidadEmisora;
    

    /**
     * Constructor
     * 
     */
    public DatosConsultaDetalles() {
        super();
        // TODO Auto-generated constructor stub
    }
    
    
    
    
    
    
    
    /**
     * Constructor
     * @param tipoTransaccion
     * @param cveUsuario
     * @param password
     * @param direccionIp
     * @param nombre
     * @param primerApellido
     * @param fechaNacimiento
     * @param segundoApellido
     * @param sexo
     * @param cveAlfaEntFedNac
     * @param cveEntidadEmisora
     */
    public DatosConsultaDetalles(int tipoTransaccion, String cveUsuario,
            String password, String direccionIp, String nombre,
            String primerApellido, String fechaNacimiento,
            String segundoApellido, String sexo, String cveAlfaEntFedNac,
            String cveEntidadEmisora) {
        this.tipoTransaccion = tipoTransaccion;
        this.cveUsuario = cveUsuario;
        this.password = password;
        this.direccionIp = direccionIp;
        this.nombre = nombre;
        this.primerApellido = primerApellido;
        this.fechaNacimiento = fechaNacimiento;
        this.segundoApellido = segundoApellido;
        this.sexo = sexo;
        this.cveAlfaEntFedNac = cveAlfaEntFedNac;
        this.cveEntidadEmisora = cveEntidadEmisora;
    }
    
    
    
    
    /**
     * Constructor sin entidad emisora
     * @param tipoTransaccion
     * @param cveUsuario
     * @param password
     * @param direccionIp
     * @param nombre
     * @param primerApellido
     * @param fechaNacimiento
     * @param segundoApellido
     * @param sexo
     * @param cveAlfaEntFedNac
     */
    public DatosConsultaDetalles(int tipoTransaccion, String cveUsuario,
            String password, String direccionIp, String nombre,
            String primerApellido, String fechaNacimiento,
            String segundoApellido, String sexo, String cveAlfaEntFedNac) {
        this.tipoTransaccion = tipoTransaccion;
        this.cveUsuario = cveUsuario;
        this.password = password;
        this.direccionIp = direccionIp;
        this.nombre = nombre;
        this.primerApellido = primerApellido;
        this.fechaNacimiento = fechaNacimiento;
        this.segundoApellido = segundoApellido;
        this.sexo = sexo;
        this.cveAlfaEntFedNac = cveAlfaEntFedNac;
    }
    /**
     * @return Obtiene el valor del atributo cveEntidadEmisora.
     */
    public String getCveEntidadEmisora() {
        return cveEntidadEmisora;
    }
    /**
     * @param cveEntidadEmisora Asigna valor al atributo cveEntidadEmisora.
     */
    public void setCveEntidadEmisora(String cveEntidadEmisora) {
        this.cveEntidadEmisora = cveEntidadEmisora;
    }
    /**
     * @return Obtiene el valor del atributo cveAlfaEntFedNac.
     */
    public String getCveAlfaEntFedNac() {
        return cveAlfaEntFedNac;
    }
    /**
     * @param cveAlfaEntFedNac Asigna valor al atributo cveAlfaEntFedNac.
     */
    public void setCveAlfaEntFedNac(String cveAlfaEntFedNac) {
        this.cveAlfaEntFedNac = cveAlfaEntFedNac;
    }
    /**
     * @return Obtiene el valor del atributo cveUsuario.
     */
    public String getCveUsuario() {
        return cveUsuario;
    }
    /**
     * @param cveUsuario Asigna valor al atributo cveUsuario.
     */
    public void setCveUsuario(String cveUsuario) {
        this.cveUsuario = cveUsuario;
    }
    /**
     * @return Obtiene el valor del atributo direccionIp.
     */
    public String getDireccionIp() {
        return direccionIp;
    }
    /**
     * @param direccionIp Asigna valor al atributo direccionIp.
     */
    public void setDireccionIp(String direccionIp) {
        this.direccionIp = direccionIp;
    }
    /**
     * @return Obtiene el valor del atributo fechaNacimiento.
     */
    public String getFechaNacimiento() {
        return fechaNacimiento;
    }
    /**
     * @param fechaNacimiento Asigna valor al atributo fechaNacimiento.
     */
    public void setFechaNacimiento(String fechaNacimiento) {
        this.fechaNacimiento = fechaNacimiento;
    }
    /**
     * @return Obtiene el valor del atributo nombre.
     */
    public String getNombre() {
        return nombre;
    }
    /**
     * @param nombre Asigna valor al atributo nombre.
     */
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
    /**
     * @return Obtiene el valor del atributo password.
     */
    public String getPassword() {
        return password;
    }
    /**
     * @param password Asigna valor al atributo password.
     */
    public void setPassword(String password) {
        this.password = password;
    }
    /**
     * @return Obtiene el valor del atributo primerApellido.
     */
    public String getPrimerApellido() {
        return primerApellido;
    }
    /**
     * @param primerApellido Asigna valor al atributo primerApellido.
     */
    public void setPrimerApellido(String primerApellido) {
        this.primerApellido = primerApellido;
    }
    /**
     * @return Obtiene el valor del atributo segundoApellido.
     */
    public String getSegundoApellido() {
        return segundoApellido;
    }
    /**
     * @param segundoApellido Asigna valor al atributo segundoApellido.
     */
    public void setSegundoApellido(String segundoApellido) {
        this.segundoApellido = segundoApellido;
    }
    /**
     * @return Obtiene el valor del atributo sexo.
     */
    public String getSexo() {
        return sexo;
    }
    /**
     * @param sexo Asigna valor al atributo sexo.
     */
    public void setSexo(String sexo) {
        this.sexo = sexo;
    }
    /**
     * @return Obtiene el valor del atributo tipoTransaccion.
     */
    public int getTipoTransaccion() {
        return tipoTransaccion;
    }
    /**
     * @param tipoTransaccion Asigna valor al atributo tipoTransaccion.
     */
    public void setTipoTransaccion(int tipoTransaccion) {
        this.tipoTransaccion = tipoTransaccion;
    }
}// end class

File: curp_ws/clientesws/WsCurpConsultaClient/DatosConsultaDetalles.rb
class DatosConsultaCurp
    def initialize(direccion_ip=nil,tipo_transaccion=nil,cve_usuario=nil,password=nil,cve_curp=nil,cve_entidad_emisora=nil,nombre=nil,primer_apellido=nil,segundo_apellido=nil,fecha_nacimiento=nil,sexo=nil,cve_alfa_ent_fed_nac=nil)
        @direccion_ip=direccion_ip
        @tipo_transaccion=tipo_transaccion
        @usuario=usuario
        @password=password
        @cve_curp=cve_curp
        @cve_entidad_emisora=cve_entidad_emisora
        @nombre=nombre
        @primer_apellido=primer_apellido
        @segundo_apellido=segundo_apellido
        @fecha_nacimiento=fecha_nacimiento
        @sexo=sexo
        @cve_alfa_ent_fed_nac=cve_alfa_ent_fed_nac
    end
    #Setters
    def set_direccion_ip direccion_ip
        @direccion_ip=direccion_ip
    end
    def set_tipo_transaccion tipo_transaccion
        @tipo_transaccion=tipo_transaccion
    end
    def set_usuario usuario
        @usuario=usuario
    end
    def set_password password
        @password=password
    end
    def set_cve_curp cve_curp
        @cve_curp=cve_curp
    end
    def set_cve_entidad_emisora cve_entidad_emisora
        @cve_entidad_emisora=cve_entidad_emisora
    end
    def set_nombre nombre
        @nombre=nombre
    end
    def set_primer_apellido primer_apellido
        @primer_apellido=primer_apellido
    end
    def set_segundo_apellido segundo_apellido
        @segundo_apellido=segundo_apellido
    end
    def set_fecha_nacimiento fecha_nacimiento
        @fecha_nacimiento=fecha_nacimiento
    end
    def set_sexo sexo
        @sexo=sexo
    end
    def set_cve_alfa_ent_fed_nac cve_alfa_ent_fed_nac
        @cve_alfa_ent_fed_nac=cve_alfa_ent_fed_nac
    end
    #Getters
    def get_direccion_ip
        @direccion_ip
    end
    def get_tipo_transaccion
        @tipo_transaccion
    end
    def get_usuario
        @usuario
    end
    def get_password
        @password
    end
    def get_cve_curp
        @cve_curp
    end
    def get_cve_entidad_emisora
        @cve_entidad_emisora
    end
    def get_nombre
        @nombre
    end
    def get_primer_apellido
        @primer_apellido
    end
    def get_segundo_apellido
        @segundo_apellido
    end
    def get_fecha_nacimiento        @fecha_nacimiento
    end
    def get_sexo
        @sexo
    end
    def get_cve_alfa_ent_fed_nac
        @cve_alfa_ent_fed_nac
    end
end

